source/sundials/arkode.cc

```diff
@@ -234,8 +240,326 @@ namespace SUNDIALS
 
       return err;
     }
+#  else
+
+    template <typename VectorType>
+    int
+    t_arkode_jac_times_vec_function(N_Vector v,
+                                    N_Vector Jv,
+                                    realtype t,
+                                    N_Vector y,
+                                    N_Vector fy,
+                                    void *   user_data,
+                                    N_Vector)
+    {
+      ARKode<VectorType> &solver =
+        *static_cast<ARKode<VectorType> *>(user_data);
```

Comment:
We can't `dynamic_cast` a `void *`? Instead I added an assert for not `nullptr` which already saved me once during debbugging;)

@peterrum Should it be `Assert` or `AssertThrow`?  Since it is an internal error I would go for `Assert`. The program will then segfault in release mode.

Reply:
This looks like an internal function!? So it should  a simple assert. AssertThrow is normally used for checking user input.